src/main/java/org/opensearch/security/dlic/rest/validation/CredentialsValidator.java

```diff
@@ -99,4 +111,53 @@ public boolean validate() {
         }
         return true;
     }
+
+    private boolean regexBasedPasswordValidation(final String regex, final String password) {
+        // Password can be null for an existing user. Regex will validate password if present
+        if (!Pattern.compile("^" + regex + "$").matcher(password).matches()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Regex does not match password");
+            }
+            this.errorType = ErrorType.INVALID_PASSWORD;
+            return false;
+        }
+        return true;
+    }
+
+    private boolean scoreBasedPasswordValidation(final PasswordValidationScoreStrength scoreStrength,
```

Comment:
Thanks for the API tests, but please also add unit tests that can verify the code all paths of this method.